### ISSUE TITLE
Difficult to see checkboxes within a table in Dark Mode

### ISSUE THREAD

**Issue by jamesobutler** [Slicer/Slicer#5185]
Originally from https://discourse.slicer.org/t/selection-boxes-in-save-dialog-box-is-invisible-in-dark-mode/13442

![image](https://user-images.githubusercontent.com/15837524/93033960-73b99400-f606-11ea-99e7-c7220d940788.png)

It appears in light mode that border of the checkbox is #c8c8c8 and based on the values specified for Dark Mode, this color appears to be defined by QPalette::Mid (see below). 

**Slicer Light**
Oddly in the Slicer light style Mid and Light are the same value when they shouldn’t be. Read about QPalette [ColorRoles](https://doc.qt.io/qt-5/qpalette.html#ColorRole-enum).
https://github.com/Slicer/Slicer/blob/c565268cf68c09b54e0e65fbfed5bd5faa404401/Base/QTGUI/qSlicerStyle.cxx#L228-L232
**Slicer Dark**
https://github.com/Slicer/Slicer/blob/c565268cf68c09b54e0e65fbfed5bd5faa404401/Base/QTGUI/qSlicerStyle.cxx#L264-L268

It is harder to see in dark mode because QPalette::Mid is closer to QPalette::Base(the filled color), than it is in light mode. This is a weird case of a widget (QCheckbox) that is using the color of QPalette::Base even though it is within another widget (QTableWidget) which also uses QPalette::Base by default for its contents. Setting it to use the QPalette::AlternateBase could potentially work assuming that this widget continues to not use alternatingRowColors.

**Comment by AaronGee1** [comment]
Can I take this issue? 

Is this ticket just to fix the dark mode button appearence? Or did we want to address the use of same color value for light mode light/mid or is this an intended style for the GUI?

**Comment by AaronGee1** [comment]
I did some digging and it turns out theres two types of checkboxs in the table for filename and options. I don't think the checkbox in filename is actually a QCheckbox but an QAbstractItemView indicator and it looks like its unfilled so it just takes the color of the table. I found two ways to address the issue using either a stylesheet or editing the palette.

1) By editing the palette we can change the background of the item in the table so the indicator box will show up and we can simply change the QCheckBox base palette to change the fill color of its checkbox.

![DarkModePalette](https://user-images.githubusercontent.com/19867990/172054242-9d7c07f7-5658-4722-a274-52cf8233a6e8.png)

![NormalPalette](https://user-images.githubusercontent.com/19867990/172054292-cec771ed-2e0f-459a-934f-1fe22cf4f231.png)

2) Alternatively we can set a stylesheet on table creation to change the color of unchecked boxes to alternate base. I think this option looks the best but using stylesheets seems to introduce other problems. The style doesn't update properly for the table when changing the apperance in the settings and requires a restart to set the colors correctly.

![StylesheetDarkMode](https://user-images.githubusercontent.com/19867990/172054377-67bb0bf0-7f9d-45f7-98fb-2e4235ecf990.png)

![Stylesheet](https://user-images.githubusercontent.com/19867990/172054383-ccd051fc-722b-42d4-a55e-433967e241da.png)
